### DISCUSSION TITLE
about class Ref<T> { public T Value; }

### DISCUSSION THREAD

**Discussion by vanasis** [dadhi/DryIoc#596] [Q&A] [answered]
what is usage of "class Ref<T> { public T Value; }"

**Comment by dadhi** [comment, marked as answer]
@vanasis The `Ref` is the helper abstraction to allow optimistic-concurrency Swap operation on the `Value`.
I am using it internally in DryIoc in concatenation with the immutable collections like `ImHashMap` to enable thread-safe set of the new value of the map **without locking** (this is the whole purpose of the optimistic concurrency). You may look through the `Ref.Swap` or just a `.Swap` method usages in `Container.cs` for the examples.
